rosbag2_transport/src/rosbag2_transport/rosbag2_transport_python.cpp

```diff
@@ -124,6 +128,32 @@ rosbag2_transport_play(PyObject * Py_UNUSED(self), PyObject * args, PyObject * k
   play_options.node_prefix = std::string(node_prefix);
   play_options.read_ahead_queue_size = read_ahead_queue_size;
 
+  std::set<std::string> exclude_topics_list;
+  if (exclude_topics) {
+    PyObject * topic_exclude_iterator = PyObject_GetIter(exclude_topics);
+    if (topic_exclude_iterator != nullptr) {
+      PyObject * topic;
+      while ((topic = PyIter_Next(topic_exclude_iterator))) {
+        exclude_topics_list.insert(PyUnicode_AsUTF8(topic));
+
+        Py_DECREF(topic);
+      }
+      Py_DECREF(topic_exclude_iterator);
+    }
+
+    if (exclude_topics_list.empty() == false) {
+      auto topic_filter_function = [exclude_topics_list](const std::string& topic)
+      {
+        auto entry = exclude_topics_list.find(topic);
```

Comment:
Could this block just be replaced by 
```return (exclude_topics_list.find(topic) != exclude_topics_list.end());```

Reply:
I would argue, that the current implementation is more readable and therefore more expressive than putting everything into one line. However, if you like to change this, I'll do it.

Reply:
I thought to believe that it's easier for the compiler to inline and optimize this as there is no if and thus no branch prediction and stuff might happen. But it's solely premature optimization thinking :)